staker/staker.go

```diff
@@ -332,6 +339,17 @@ func NewStaker(
 		if err != nil {
 			return nil, err
```

Comment:
I'm a bit worried we'll end up hitting this when the fast confirmer isn't a safe, but is actually another validator. There are two cases here:
- The fast confirmer is an EOA: before we call NewFastConfirmSafe, we should check if the fast confirmer's code is empty (I think there's a CodeAt method on the l1 client or something similar). If it's an EOA, and it's not our address, disable fast confirmations.
- The fast confirmer is someone else's validator wallet: We should test the error returned from NewFastConfirmSafe with `headerreader.ExecutionRevertedRegexp` . If it matches, then we should disable fast confirmations, because it doesn't seem to be a safe.
- We should also wrap these errors indicating where they come from, e.g. `return nil, fmt.Errorf("loading fast confirm safe: %w", err)` so it isn't just a generic "execution reverted" error

Reply:
Done.